float32[] quat
float32[] pos
---
float32[] ik_sols
float32[] ik_ranks